### PR TITLE
task-3

### PR DESCRIPTION
**Pull request by 1van0v**
1. What was done?
    Service is done and integrated with FE
    Additional scope:
      - Async/await
      - ES6 modules
      - webpack
      - error handlers
      - swagger
      - configure front
      - unit-tests
2. Link to product-service API - https://5a93nu96k4.execute-api.eu-west-1.amazonaws.com/products
3. Link to FE MR (YOUR OWN REPOSITORY) - https://github.com/1van0v/shop-react-redux-cloudfront/pull/2
4. Link to deployed FE - http://product-front-end.s3-website-eu-west-1.amazonaws.com/
5. Link to SWAGGER - https://app.swaggerhub.com/apis-docs/1van0v/dev-product_service/1#/default/get_products

### CODE REVIEW

File: product-service/package.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "product-service",
+  "version": "1.0.0",
+  "description": "Serverless webpack example using ecma script",
+  "scripts": {
+    "test": "jest --coverage"
+  },
+  "devDependencies": {
+    "@babel/core": "^7.11.1",
```
Comment (ragexe) [inline product-service/package.json:9]: It looks like you forgot to add the '@babel/plugin-transform-runtime' package to the dev dependencies. Without installing this package on my machine all of the tests are failed with an error 
```
Cannot find module '@babel/plugin-transform-runtime'
```

File: product-service/serverless.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+service: product-service
+frameworkVersion: '2'
+
+# Add the serverless-webpack plugin
+plugins:
+  - serverless-webpack
+
+provider:
+  name: aws
+  runtime: nodejs12.x
```
Comment (ragexe) [inline product-service/serverless.yml:10]: FYI
`nodejs14.x` would be better 😄

File: product-service/package.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "product-service",
+  "version": "1.0.0",
+  "description": "Serverless webpack example using ecma script",
+  "scripts": {
+    "test": "jest --coverage"
```
Comment (ragexe) [inline product-service/package.json:6]: It would be great to add AWS deploy script.

File: product-service/serverless.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  lambdaHashingVersion: 20201221
+  region: eu-west-1
+
+functions:
+  getProductsList:
+    handler: handlers/products-list.getProductsList
+    events:
+      - httpApi:
+          method: get
+          path: /products
```
Comment (ragexe) [inline product-service/serverless.yml:20]: The Acceptance Criteria specified the names of the endpoints as `/getProductsList'. Please try to follow the requirements of the assignments exactly, it is quite important.

File: product-service/tests/utils/create-json-response.spec.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+      'Access-Control-Allow-Origin': '*',
+      'Access-Control-Allow-Credentials': true,
+      'Content-Type': 'application/json'
+    },
+    statusCode: 200,
+  })
+})
+
+it('should set specified status code', () => {
+  expect(createJsonResponse({ test: 123 }, 404)).toEqual({
```
Comment (ragexe) [inline product-service/tests/utils/create-json-response.spec.js:16]: Seems that your lambdas return 404 status code only in test cases
image

File: product-service/serverless.yml (hunk 4 above)
Comment (1van0v) [reply product-service/serverless.yml:20]: thanks for the review. But, as I can see, the endpoint should be `/products`. Here is a screenshot with task 
image

**Pull request merged**